### DISCUSSION TITLE
mariadb on Macbook M1 - macOS 11.1

### DISCUSSION THREAD

**Discussion by thomasriise** [Homebrew/discussions#503] [Everyday usage] [answered]
Hi guys - trying to install mariadb on big sur 11.1 - on the new Macbook Pro M1.

```
~$ brew install mariadb
Error: mariadb: no bottle available!
You can try to install from source with:
  brew install --build-from-source mariadb
Please note building from source is unsupported. You will encounter build
failures with some formulae. If you experience any issues please create pull
requests instead of asking for help on Homebrew's GitHub, Twitter or any other
official channels.
```

If I try to build from source, this happens:

```
Last 15 lines from /Users/tr/Library/Logs/Homebrew/mariadb/02.make:
[ 93%] Built target pcre2-8
Scanning dependencies of target pcre2-posix
[ 96%] Building C object CMakeFiles/pcre2-posix.dir/src/pcre2posix.c.o
[100%] Linking C static library libpcre2-posix.a
[100%] Built target pcre2-posix
cd /tmp/mariadb-20210111-47761-ubil5d/mariadb-10.5.8/extra/pcre2/src/pcre2-build && /opt/homebrew/Cellar/cmake/3.19.2/bin/cmake -E touch /tmp/mariadb-20210111-47761-ubil5d/mariadb-10.5.8/extra/pcre2/src/pcre2-stamp/pcre2-build
[ 10%] No install step for 'pcre2'
cd /tmp/mariadb-20210111-47761-ubil5d/mariadb-10.5.8/extra/pcre2/src/pcre2-build && /opt/homebrew/Cellar/cmake/3.19.2/bin/cmake -E echo_append
cd /tmp/mariadb-20210111-47761-ubil5d/mariadb-10.5.8/extra/pcre2/src/pcre2-build && /opt/homebrew/Cellar/cmake/3.19.2/bin/cmake -E touch /tmp/mariadb-20210111-47761-ubil5d/mariadb-10.5.8/extra/pcre2/src/pcre2-stamp/pcre2-install
[ 10%] Completed 'pcre2'
/opt/homebrew/Cellar/cmake/3.19.2/bin/cmake -E make_directory /tmp/mariadb-20210111-47761-ubil5d/mariadb-10.5.8/CMakeFiles
/opt/homebrew/Cellar/cmake/3.19.2/bin/cmake -E touch /tmp/mariadb-20210111-47761-ubil5d/mariadb-10.5.8/CMakeFiles/pcre2-complete
/opt/homebrew/Cellar/cmake/3.19.2/bin/cmake -E touch /tmp/mariadb-20210111-47761-ubil5d/mariadb-10.5.8/extra/pcre2/src/pcre2-stamp/pcre2-done
[ 10%] Built target pcre2
make: *** [all] Error 2
```

Mysql works fine and so does all other bottles - any ideas?

**Comment by SMillerDev** [comment, marked as answer]
> Mysql works fine and so does all other bottles - any ideas?

MariaDB doesn't yet, you'll have to wait for it to be supported or build it from source as the message says.

**Reply by ugr** [reply]
There is an invalid preprocessor check in a header file (the message is in the full log file 02.make). Obviously, the target platform cannot be correctly detected, so that an #error directive stops the compilation.
There exist some patches for this issue. The macports project successfully compiles mariadb-10.5-server for M1, and its working fine. But I don't know if the patches are official ones from the MariaDB developers, or 'special' patches for the macports project.

**Reply by cxpreng** [reply]
Any idea why all ports of mariadb-server after 10.2 are betas?

**Reply by gromgit** [reply]
They aren't, as far as I can tell, and how is this related to Homebrew?

**Comment by thomasriise** [comment]
Thanks @SMillerDev - now I saw some packages has the "arm64_big_sur" tag - and mariadb does not. My bad, sorry for that.

**Comment by chobotnice007** [comment]
Use MySQL. There isn't any way to run MariaDB on Apple Silicon. @thomasriise

**Reply by thomasriise** [reply]
👍

**Comment by tholu** [comment]
Fix will be available shortly, check out https://github.com/Homebrew/homebrew-core/pull/70060

**Reply by tholu** [reply]
@thomasriise It's available now.

**Reply by thomasriise** [reply]
Awesome! Thanks all!

**Comment by ajaxray** [comment]
Already fixed. Use the following command.
```
arch -x86_64 brew install mariadb
```

I've just installed it :D

**Reply by carlocab** [reply]
That installs a non-native version of `mariadb`. `mariadb` [has an ARM bottle now](https://formulae.brew.sh/formula/mariadb#default), so you may want to install a native version instead.

**Reply by ajaxray** [reply]
@carlocab Thanks for the information. 👍🏼

But I am not being able to install it with any of the following 
```
brew install mariadb
brew install mariadb@10.5
brew install mariadb@10.5.8
```

Can you please help?

**Reply by carlocab** [reply]
What does `brew doctor` and `brew config` say?

**Reply by androidformax** [reply]
I try install mariadb on Mac M1 - on all commands I have error:
Error: Cannot install under Rosetta 2 in ARM default prefix (/opt/homebrew)!

Did anybody know how to install mariadb to Macbook Air on M1 chip ?

**Reply by SMillerDev** [reply]
Don't run the terminal with Rosetta enabled and it'll work.

**Comment by ZidCode** [comment]
Hi, did you achieve to run MariaDB on arm64? 

`brew services restart mariadb` shows that mariadb sucessfully started but the service is not running. Does have anyone an idea?

**Reply by Avarsa** [reply]
Hi, any fix to this one for you?

**Comment by thomasriise** [comment]
Yep - worked like a charm.

 --  Thomas
